cvat/apps/engine/media_extractors.py

```diff
@@ -179,25 +194,48 @@ def _make_name():
 
 class ZipReader(ImageListReader):
     def __init__(self, source_path, step=1, start=0, stop=None):
-        self._zip_source = zipfile.ZipFile(source_path[0], mode='r')
+        self._zip_source = zipfile.ZipFile(source_path[0], mode='a')
         self.extract_dir = source_path[1] if len(source_path) > 1 else None
         file_list = [f for f in self._zip_source.namelist() if get_mime(f) == 'image']
         super().__init__(file_list, step, start, stop)
 
     def __del__(self):
         self._zip_source.close()
 
-    def get_preview(self):
+    def get_preview(self, dimension="2d"):
+        if dimension == "3d":
+            fp = open(os.path.join(os.path.dirname(__file__), 'assets/3d_preview.jpeg'), "rb")
+            return self._get_preview(fp)
         io_image = io.BytesIO(self._zip_source.read(self._source_path[0]))
         return self._get_preview(io_image)
 
-    def get_image_size(self, i):
+    def get_image_size(self, i, dimension="2d"):
+        if dimension == "3d":
+            with self._zip_source.open(self._source_path[i], "r") as file:
+                properties = get_pcd_properties(file)
+                return int(properties["WIDTH"]),  int(properties["HEIGHT"])
         img = Image.open(io.BytesIO(self._zip_source.read(self._source_path[i])))
         return img.width, img.height
 
     def get_image(self, i):
         return io.BytesIO(self._zip_source.read(self._source_path[i]))
 
+    def add_files(self, source_path):
+        root_path = os.path.split(self._zip_source.filename)[0]
+        for path in source_path:
+            self._zip_source.write(path, path.replace(root_path, ""))
+
+    def get_zip_filename(self):
+        return self._zip_source.filename
+
+    def initialize_for_3d(self, source_path, step=1, start=0, stop=None):
```

Comment:
If you inherit a new class, this method is no longer needed

Reply:
I agree. Instead of multiple "if" in the class, let us create a new one like LidarDataReader or something like that.

Reply:
Image dataset and pcd dataset can be in a zip format, to classify if the dataset is for a 2D task or 3D task.
If we create another class as "LidarDataReader", we need to unzip the zip file twice, once for extracting the data in "ZipReader",then again in "LidarReader" class since zip is a common media-type.

https://github.com/openvinotoolkit/cvat/blob/0877de4885232cdfa7d6b8e3d9516a756df06f99/cvat/apps/engine/media_extractors.py#L545

Instead we modified the "ZipReader" class to and used reconcile method to distinguish between the dimensionType and now the multiple if's are not present.
It is only checked once on initialisation.